pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -85,34 +106,102 @@ func (l *PodInterface) Plug(vmi *v1.VirtualMachineInstance, iface *v1.Interface,
 		}
 
 		if err := driver.preparePodNetworkInterfaces(); err != nil {
-			log.Log.Reason(err).Critical("failed to prepared pod networking")
+			log.Log.Reason(err).Critical("failed to prepare pod networking")
 			panic(err)
 		}
 
-		// After the network is configured, cache the result
-		// in case this function is called again.
-		err = driver.decorateConfig()
+		err = driver.setCachedInterface(pidStr, iface.Name)
 		if err != nil {
-			log.Log.Reason(err).Critical("failed to create libvirt configuration")
+			log.Log.Reason(err).Critical("failed to save interface configuration")
 			panic(err)
 		}
 
-		err = driver.setCachedInterface(iface.Name)
+		err = driver.setCachedVIF(pidStr, iface.Name)
 		if err != nil {
-			log.Log.Reason(err).Critical("failed to save interface configuration")
+			log.Log.Reason(err).Critical("failed to save vif configuration")
 			panic(err)
 		}
 	}
 
 	return nil
 }
 
-func getBinding(vmi *v1.VirtualMachineInstance, iface *v1.Interface, network *v1.Network, domain *api.Domain, podInterfaceName string) (BindMechanism, error) {
-	podInterfaceNum, err := findInterfaceByName(domain.Spec.Devices.Interfaces, iface.Name)
+func ensureDHCP(vmi *v1.VirtualMachineInstance, driver BindMechanism, podInterfaceName string) error {
+	dhcpStartedFile := fmt.Sprintf("/var/run/kubevirt-private/dhcp_started-%s", podInterfaceName)
```

Comment:
This part isn't very clear to me...
Will this file be cleaned after `panic`?
If not, it will mean that we will always have to make sure that no other method can panic()  after DHCP is configured.
Or am I not understanding it correctly?

Reply:
Yes, DHCP is the only step that is still in phase2 (launcher). This file is created inside launcher private storage and AFAIU will not survive panic.

Reply:
Sounds good. :) 
I guess I wasn't really sure why do we need the file and won't simply panic here, but I'm fine with it either way.

Reply:
phase2 may be called multiple times, so attempting to start DHCP multiple times for the same pod is not ideal. I don't think it would succeed to do it because of port occupied by the first server started; the error will be silently swallowed because of how we start and manage the DHCP thread; but it's better to avoid it.